### ISSUE TITLE
Azure Functions Python Runtime 5/25/2019 [breaking]

### ISSUE THREAD

**Issue by maiqbal11** [Azure/azure-functions-core-tools#1314]
We are planning to release a breaking change affecting Python function apps in the third week of May 2019. 
 
**TL;DR**

**You will need azure-functions-core-tools version 2.5.553 or greater for this to work**. If you previously published your function app using the flag `--build-native-deps` and do not want your function app to break, you will need to re-publish using `func azure functionapp publish <app_name> --build-native-deps`. 
 
**What's changing?**

We are making changes to our Python app runtime flow to remove support for `PyInstaller` which we were previously using to compress package size. PR to remove PyInstaller packaging: https://github.com/Azure/azure-functions-core-tools/pull/1152.  

**Rationale**
1. The version of azure-functions-worker (https://pypi.org/project/azure-functions-worker/) needs to be controlled by Azure Functions and was previously being bundled with user dependencies. This gives the dev team less control over the worker code that we are running and any change on our side also requires the user to republish their app. As a result, any improvements we roll out are not transparent to the user and they have to be concerned about republishing to get the latest bits.
2. Future performance improvements require that we change the runtime flow. Since we cannot make assumptions about what worker version the user is currently bundling, we are not able to make changes consistently across the board.
3. `PyInstaller` does not support all packages (https://github.com/pyinstaller/pyinstaller/wiki/Supported-Packages). We have improved the publishing flow so that all user scenarios will be supported out of the box.

Please use https://github.com/Azure/app-service-announcements-discussions/issues/91 for discussion related to this issue.
